### PR TITLE
[WIP] Web app

### PR DESCRIPTION
**Pull request by joecorcoran**
**Not ready to merge**

Doesn't do much and isn't very robust, but here we go... :rocket:

To do:

- [x] create `travis-yml-staging`
- [x] create `travis-yml-production`
- [x] create `travis-pro-yml-staging`
- [x] create `travis-pro-yml-production`
- [x] Handle parse failures
- [x] Add messages to response
- [x] Consider security risks
  - Public requests
  - DoS
- [x] Consider implications of downtime
  - When app is slow
  - When app has crashed
  - When app is deploying
- [x] set up slack deployments
- [ ] Drain logs to Papertrail (app needs to be deployed)
- [x] Sentry
- [ ] Consider production Puma config
- [x] Force SSL
- [x] DNS
- [x] Basic versioning
- [x] keychain file/s (auth_keys (array of keys apps can use. see `trials`))

### CODE REVIEW

File: lib/travis/yaml/web/config.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require 'travis/config'
+
+module Travis::Yaml::Web
+  class Config < Travis::Config
+    define auth_keys: ['abc123']
```
Comment (carlad) [inline lib/travis/yaml/web/config.rb]: Config for `auth_keys` has been added to keychain and trvs-uploaded to (for the moment) the org-staging app so am thinking this needs to change? Or is this just setting up a placeholder for local/dev work...?

Comment (joecorcoran) [reply lib/travis/yaml/web/config.rb]: That only gets loaded if the keychain version isn't found. 👍

**Comment by keradus** [comment]
PR Summary:
> WIP (...)

PR Description:
> Not ready to merge

was it intended to be merged @carlad ?

**Comment by carlad** [comment]
@keradus sorry for the confusion. It was ready to merge but I failed to update the title. Thanks @joecorcoran for looking after that :smile:

**Pull request merged**